### PR TITLE
default to level-zero, add CHIP_VERSION

### PR DESCRIPTION
**Pull request by pvelesko**
* When both Level Zero and OpenCL are available, default to Level Zero
* Add `CHIP_VERSION` to `chipStarConfig.hh` @Sarbojit2019

### CODE REVIEW

**Comment by linehill** [comment]
> Add CHIP_VERSION to chipStarConfig.hh @Sarbojit2019

I'm curious about this. What is the use case for this?

**Comment by Sarbojit2019** [comment]
@linehill
I need it for Interop handling. oneAPI 2024.0.0 has changed make_queue interface. To handle changed API signature chipStar needs additional changes. To handle different possible combinations between oneAPI and chipStar better I need the interface to query chipStar runtime version. For more information regarding the ask please check the ticket #694 .

BTW, I think in any case we should have a runtime query interface to know the chipStar version. That is very primitive requirement to me.

**Comment by linehill** [comment]
> BTW, I think in any case we should have a runtime query interface to know the chipStar version. 

Makes sense to me. Applications might load a different version of chipStar.

File: CMakeLists.txt [hunk 1]
```diff
@@ -35,8 +35,9 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-duplicate-decl-specifier \
 cmake_minimum_required(VERSION 3.20 FATAL_ERROR)
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/cmake")
 
+set(CHIPSTAR_VERSION 1.1)
```
Comment (linehill) [inline CMakeLists.txt:38]: Let's bump this up so applications may tell release 1.1 and main branch (aka. the next release) apart.
```suggestion
set(CHIPSTAR_VERSION 1.2)
```

**Comment by pvelesko** [comment]
@linehill went back to 1.1 since this needs to be merged into 1.1 Release because of the default

**Pull request merged**